contracts/lbp/LBPWrapper.sol

```diff
@@ -33,104 +36,100 @@ contract LBPWrapper {
-            address(this),
-            _swapEnabledOnStart
+        lbp = ILBP(
+            ILBPFactory(_LBPFactory).create(
+                _name,
+                _symbol,
+                _tokens,
+                _weights,
+                SWAP_FEE_PERCENTAGE,
+                address(this),
+                false // setSwapEnabled at pool creation to false
+            )
         );
 
-        // to handle stack overflow
-        {
-            ILBP(lbp).updateWeightsGradually(_startTime, _endTime, _endWeights);
+        lbp.updateWeightsGradually(_startTime, _endTime, _endWeights);
+
+        return address(lbp);
+    }
+
+    /**
+     * @dev                          approve tokens for the vault and join pool to provide liquidity
+     * @param _tokens                array of tokens sorted for the LBP
+     * @param _amounts               amount of tokens to add to the pool to provide liquidity
+     * @param _receiver              address who will receive the Balancer Pool Tokens (BPT)
+     * @param _fromInternalBalance   fund tokens from the internal user balance
+     * @param _userData              userData specifies the type of join
+     */
+    function fundPool(
+        IERC20[] memory _tokens,
+        uint256[] memory _amounts,
+        address _receiver,
+        bool _fromInternalBalance,
+        bytes memory _userData
+    ) external onlyOwner {
+        require(!isPoolFunded, "LBPWrapper: pool has already been funded");
+
+        setPause(false); // unpauses the pool to add funds
+
+        IVault vault = lbp.getVault();
+        for (uint8 i; i < _tokens.length; i++) {
+            _tokens[i].approve(address(vault), _amounts[i]);
         }
 
         IVault.JoinPoolRequest memory request = IVault.JoinPoolRequest({
             maxAmountsIn: _amounts,
             userData: _userData,
-            fromInternalBalance: false,
+            fromInternalBalance: _fromInternalBalance,
             assets: _tokens
         });
 
-        // // to handle stack overflow
-        {
-            address vault = address(LBPFactory.getVault());
-            IVault(vault).joinPool(
-                ILBP(lbp).getPoolId(),
-                address(this),
-                _admin,
-                request
-            );
-        }
+        vault.joinPool(lbp.getPoolId(), address(this), _receiver, request);
 
-        return lbp;
+        isPoolFunded = true;
     }
 
     /**
-     * @dev                       set new swap fee percentage
-     * @param _swapFeePercentage  swap fee percentage
+     * @dev                     can pause/unpause trading
+     * @param _isPaused         enable/disable swapping
      */
-    function setSwapFeePercentage(uint256 _swapFeePercentage) public onlyOwner {
-        swapFeePercentage = _swapFeePercentage;
+    function setPause(bool _isPaused) public onlyOwner {
+        isPaused = _isPaused;
+        _isPaused = _isPaused ? false : true; // setSwapEnabled requires opposite bool
```

Comment:
Cool, I didn't know there exist ternery operators in Solidity :)

Reply:
Yeah thought it would be good to use here. Also my first time checking if this would be possible in solidity